astropy/io/fits/connect.py

```diff
@@ -162,7 +163,10 @@ def read_table_fits(input, hdu=None):
 
     # TODO: deal properly with unsigned integers
 
-    for key, value, comment in table.header.cards:
+    # Read time coordinates as Time
+    hdr = FITS_time().read_time(table.header, t)
```

Comment:
Why sometimes using class methods and here a "useless" object and a method call ? 
It seems (but maybe I looked too quickly!) that all this could be simplified, two functions `fits2time` / `time2fits` (or similar) could do the job and this would be easier to understand and also faster.